packages/server/src/index.ts

```diff
@@ -67,6 +67,28 @@ export class Server {
    */
   constructor() {
     this.handlers = {};
+    this.addMulticallSupport();
+  }
+
+  /**
+   * Add IMulticallable.multicall() support
+   * https://github.com/ensdomains/ens-contracts/blob/staging/contracts/resolvers/IMulticallable.sol
+   */
+  addMulticallSupport() {
+    const self = this;
+    this.add(['function multicall(bytes[] calldata data) external returns (bytes[] memory results)'], [{
+      type: 'multicall',
+      async func(args: ethers.utils.Result, {to}: RPCCall) {
+        return [await Promise.all(args[0].map(async (data: any) => {
+          try {
+            let {status, body} = await self.call({to, data});
+            if (status === 200) return body.data;
+          } catch (err) {
```

Comment:
I think we need better error handling than this. At a minimum it should log the error in some way. Perhaps we could mirror the way that eth_call does it, and supply the error message as revert data?

Reply:
Maybe this is the wrong interface?  I noticed that [Multicallable](https://github.com/ensdomains/ens-contracts/blob/master/contracts/resolvers/Multicallable.sol) is primarily used for batch-writes, where it makes sense that a single failure would kill the transaction.  But for reads, you want to be able to probe profiles that might not be implemented.

The result could be `tuple(boolean ok, bytes[] data)[]` like Multicall3, which makes propagating the error possible.

Reply:
That would also work. I was thinking of how most clients handle `eth_call`; rather than returning an error code indicating a revert, they simply return the revert data in place of the return data, and rely on the fact that reverts have `length % 32 == 4`, while return data is typically `length % 32 == 0`.

Reply:
I now return the handler error or exception as `Error(string)`